mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/handlers/JobArtifactRouteHandlerImpl.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.master.api.akka.route.handlers;
+
+import static io.mantisrx.master.jobcluster.proto.BaseResponse.ResponseCode.CLIENT_ERROR;
+import static io.mantisrx.master.jobcluster.proto.BaseResponse.ResponseCode.SUCCESS;
+
+import io.mantisrx.master.jobcluster.proto.JobArtifactProto;
+import io.mantisrx.server.master.store.JobArtifact;
+import io.mantisrx.server.master.store.JobArtifactIDAlreadyExistsException;
+import io.mantisrx.server.master.store.MantisStorageProvider;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JobArtifactRouteHandlerImpl implements JobArtifactRouteHandler {
+    private static final Logger logger = LoggerFactory.getLogger(JobArtifactRouteHandlerImpl.class);
+
+    private final MantisStorageProvider mantisStorageProvider;
+
+    public JobArtifactRouteHandlerImpl(MantisStorageProvider mantisStorageProvider) {
+        this.mantisStorageProvider = mantisStorageProvider;
+    }
+
+    @Override
+    public CompletionStage<JobArtifactProto.ListJobArtifactsResponse> get(JobArtifactProto.ListJobArtifactsRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                final List<JobArtifact> jobArtifactList = mantisStorageProvider.listJobArtifacts();
```

Comment:
"request" var is not used?

Reply:
at the moment the request doesn't have any params since it fetches all. `request` is only used to create the response using the `responseId` .

Reply:
Got it. So I assume filtering etc. is will be added there?

Reply:
I still have to figure that out but probably yes.